### ISSUE TITLE
Lab "Test User" setup not defined

### ISSUE THREAD

**Issue by chris-utter** [MicrosoftLearning/mslearn-fabric#165]
Lab/Demo: Secure data in a data warehouse
Task: Apply dynamic data masking rules to columns in a table
Step: 4

Description of issue
Step 4 asks the learner to "Connect as a test user that’s a member of the Viewer workspace role and run the following T-SQL statement", but provides no instructions on setting up a test user or connecting as a test user. There is a second credential provided in the lab that one assumes is the test user credential.

**Comment by AngieRudduck** [comment]
@chris-utter the lab instructions are used for self-paced and ILT, so we can't reference the provided credentials from Skillable/ALH. 

@theresa-i will you please review if there's a requirement to use one vs. the other without compromising the agnostic instructions? Could this be modified in the Skillable lab addendum instructions? We may also need Skillable to explicitly name the credentials.

**Comment by chris-utter** [comment]
The test credential could be generally referred to as “User 2 in the Resources tab of the Instruction pane” or something similar. The bigger issue is a lack of instructions on how to use the test user imo.

Chris Utter
Worldwide Learning
Microsoft
________________________________

**Comment by theresa-i** [comment]
> Lab/Demo: Secure data in a data warehouse Task: Apply dynamic data masking rules to columns in a table Step: 4
> 
> Description of issue Step 4 asks the learner to "Connect as a test user that’s a member of the Viewer workspace role and run the following T-SQL statement", but provides no instructions on setting up a test user or connecting as a test user. There is a second credential provided in the lab that one assumes is the test user credential.

@chris-utter - Thanks.  I'll take a look and get back to you on this.

**Comment by VitalyMCT** [comment]
This is definitely still an issue. Thanks @chris-utter for raising it.

There are instructions at the bottom of Page 1 in Skillable:

![image](https://github.com/user-attachments/assets/228c1471-18d2-4016-8721-fda12a57fb61)

However, they are very easy to miss. The users would've gone through that page 10+ times by now doing the previous labs. They are unlikely to notice this addendum.

The accounts are indeed pre-populated on the Resources tab:

![image](https://github.com/user-attachments/assets/37867be6-8796-4a91-b784-57fd9ea294e5)

I think to complete a user-friendly solution we can add instructions after step 4:

![image](https://github.com/user-attachments/assets/284b9730-b518-4beb-bc6b-010237470c9b)

The additional instructions would integrate User 2 credentials (+ details on how to add that user in) into the instructions themselves, removing the confusion. Skillable has the macro replacement technology that can do this. This IMO will be very user-friendly and easy-to-follow.

@theresa-i Can we please have this addressed?

**Comment by VitalyMCT** [comment]
The macro integration for User2 is already done by Skillable further downstream in the instructions. So I think it will be pretty easy to plug it into the beginning (workspace access) instructions as well.

![image](https://github.com/user-attachments/assets/9d2cce55-9376-48e8-8016-e2db1f568fab)

**Comment by theresa-i** [comment]
This is resolved.  We added a section to the lab setup page listing the test user accounts and explaining what each is used for.

**Comment by VitalyMCT** [comment]
@theresa-i  Appreciate the update, however, this is still only partially addressed. For example:

![image](https://github.com/user-attachments/assets/4c4facdd-1107-4a8f-85e3-907f1bf96b59)

These instructions do not make sense, as the replacement was already made. The instructions need to be reviewed end-to-end after the macro replacements so they can be followed by the attendees.

**Comment by theresa-i** [comment]
The instructions need to be usable by both course students and students who bring their own subscriptions.  How about if we leave the user names to match what's in the labs here https://github.com/MicrosoftLearning/mslearn-fabric/blob/main/Instructions/Labs/06d-secure-data-warehouse.md and not replace them at all in the instructions for the course, leaving the students to replace them manually?  Would that work?

**Comment by VitalyMCT** [comment]
@theresa-i Yes, I think that would work well. The custom Skillable Page 1 (outside the scope of this GitHub repo) already explains the 2 accounts to be used in the lab.

As far as Page 2, i.e. the instructions in this repo:

No replacement is fine. Full replacement is fine. Partial replacement is how we get them confused :)

**Comment by theresa-i** [comment]
Ok thanks - makes sense about the partial replacement.  I'll work with Skillable to make that change.  Reopening the issue.

**Comment by theresa-i** [comment]
I've requested the change with Skillable and will provide an update once the issue is addressed.

**Comment by theresa-i** [comment]
This is fixed.  Closing.

**Comment by abhidotnet** [comment]
I am trying to complete these labs and **cannot create a second user**. I am using a trial workspace. This is the message I get on the screen.

`One or more e-mail addresses with the following domains are outside your organization: ab*****@hotmail.com`

**Comment by timhill-iress** [comment]
Like the person above - abhidotnet, I'm having trouble creating a second user.
I'm working through "Ignite Edition Challenge: Prepare for the next generation of data analytics with Microsoft Fabric", so I'm using my own subscription with a trial Fabric account. The linked documentation https://learn.microsoft.com/en-gb/fabric/get-started/give-access-workspaces isn't helping me, I'm getting the same error as abhidotnet. I'm not clear if I I should be able to create a second user or not with this setup. If I can't create a second user with my setup, could the documentation be clearer about the prerequisites.

**Comment by theresa-i** [comment]
@timhill-iress & @abhidotnet Thanks for the feedback.  Instructions for creating the second user have been added in updated instructions.  Also noted in the updated instructions is that you can perform the exercise without a second user and limit the scope of the exercise.  See:  https://learn.microsoft.com/en-us/training/modules/secure-data-warehouse-in-microsoft-fabric/6-exercise-secure-data-warehouse

@abhidotnet - the user identity has to be created in Microsoft 365 or Entra. The error indicates that you're a member of an organization so you'd need to have your team's identity admin create the user.  

@timhill-iress - User identities are created at the tenant level, then granted subscription access.  Even though you have your own subscription, you must be a member of a role that has user administration rights in the tenant where the subscription resides to create another user.  Work with your identity admin to create the user identity, grant them access to your subscription and then add them to your Fabric environment.
